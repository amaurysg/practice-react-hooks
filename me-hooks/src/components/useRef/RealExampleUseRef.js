import React, {useRef, useState} from 'react'
import MultipleCustomsHooks from '../examples/MultipleCustomsHooks'


const RealExampleUseRef = () => {

 
  const [show, setShow] = useState(false)



  return (
    <div>
        
        <h1>Real Example useRef</h1>
        <hr/>

  
        { show && <MultipleCustomsHooks /> }
        <button 
          className='btn btn-outline-primary mt-3'
          onClick = { () => {setShow(!show)}  }  > 
          true / false
         </button>
      
    </div>
  )
}

export default RealExampleUseRef
