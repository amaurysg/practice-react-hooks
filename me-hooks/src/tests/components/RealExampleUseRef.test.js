import { shallow } from "enzyme";
import RealExampleUseRef from "../../components/useRef/RealExampleUseRef";

describe("Test componet RealExampleUseRef", () => {
  const wrapper = shallow(<RealExampleUseRef />);

  test("Debe mostrarse correctamente ", () => {
    //render correct
    expect(wrapper).toMatchSnapshot();
    //here wait that MultipleCustomsHooks  not exist with false !
    expect(wrapper.find("MultipleCustomsHooks").exists()).toBe(false);
  });

  test("Debe mostrar el componente <MultipleCustomsHooks />", () => {
    //Here simulated click
    wrapper.find("button").simulate("click");

    //here wait that MultipleCustomsHooks exist with true !
    expect(wrapper.find("MultipleCustomsHooks").exists()).toBe(true);
  });
});
