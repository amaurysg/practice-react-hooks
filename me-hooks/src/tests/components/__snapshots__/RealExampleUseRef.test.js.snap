// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test componet RealExampleUseRef Debe mostrarse correctamente  1`] = `
<div>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    true / false
  </button>
</div>
`;
